clients/go/worker/jobWorker_test.go

```diff
@@ -0,0 +1,116 @@
+package worker
+
+import (
+	"fmt"
+	"github.com/golang/mock/gomock"
+	"github.com/golang/protobuf/proto"
+	"github.com/zeebe-io/zeebe/clients/go/entities"
+	"github.com/zeebe-io/zeebe/clients/go/mock_pb"
+	"github.com/zeebe-io/zeebe/clients/go/pb"
+	"github.com/zeebe-io/zeebe/clients/go/utils"
+	"testing"
+	"time"
+)
+
+// rpcMsg implements the gomock.Matcher interface
+type rpcMsg struct {
+	msg proto.Message
+}
+
+func (r *rpcMsg) Matches(msg interface{}) bool {
+	m, ok := msg.(proto.Message)
+	if !ok {
+		return false
+	}
+	return proto.Equal(m, r.msg)
+}
+
+func (r *rpcMsg) String() string {
+	return fmt.Sprintf("is %s", r.msg)
+}
+
+func TestJobWorkerActivateJobsDefault(t *testing.T) {
```

Comment:
Is this only testing that the request is made with defaults? Or is it testing that you actually receive jobs? If it's the former, shouldn't that go in a separate test? And have another test that actually verifies that you actually handle all the jobs

Reply:
The actually it should verify that we receive jobs. I know the test looks strange but I had problems with mocking this correctly otherwise as the poller will try to fetch new jobs will the test is running. So my initial test was flaky as hell